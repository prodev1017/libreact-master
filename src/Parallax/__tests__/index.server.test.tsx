/** @jest-environment node */
import {createElement as h} from 'react';
import {renderToStaticMarkup} from 'react-dom/server';
import {Parallax} from '..';

describe('<Parallax> Server', () => {
  it('exists', () => {
    expect(typeof Parallax).toBe('function');
  });

  it('renders children', () => {
    const html = renderToStaticMarkup(
      <Parallax>
        <div>foobar</div>
      </Parallax>
    );

    expect(html.includes('foobar</div>')).toBe(true);

    const html2 = renderToStaticMarkup(
      <Parallax>{() =>
        <div>foobar</div>
      }</Parallax>
    );

    expect(html2.includes('foobar</div>')).toBe(true);
  });

  it('provides expected data', () => {
    const html = renderToStaticMarkup(
      <Parallax>{(data) => {

        expect(data).toMatchObject({
          value: 0,
          distance: Infinity,
          travelled: 0
        });

        return <div>value: {data.value}</div>;
      }}</Parallax>
    );

    expect(html.includes('value: 0</div>')).toBe(true);

    renderToStaticMarkup(
      <Parallax distance={300}>{(data) => {

        expect(data).toMatchObject({
          distance: 300
        });

        return <div>value: {data.value}</div>;
      }}</Parallax>
    );
  });

  it('throws on invalid children in development mode', () => {
    const mode = process.env.NODE_ENV;

    process.env.NODE_ENV = 'development';

    const Comp = () => <div>foobar</div>;
    const render = () => renderToStaticMarkup(
      <Parallax>{() => <Comp />}</Parallax>
    );

    expect(render).toThrowErrorMatchingSnapshot();

    process.env.NODE_ENV = 'production';

    expect(render).not.toThrow();

    process.env.NODE_ENV = mode;
  });
});
