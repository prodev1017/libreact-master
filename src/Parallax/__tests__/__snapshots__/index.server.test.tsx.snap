// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Parallax> Server throws on invalid children in development mode 1`] = `"<Parallax> accepts a single child which must be a plain DOM element or a function that returns one."`;
